lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -211,57 +213,8 @@ constexpr auto local_protocol_name = "lws-everest-client";
 static const struct lws_protocols protocols[] = {{local_protocol_name, callback_minimal, 0, 0, 0, NULL, 0},
                                                  LWS_PROTOCOL_LIST_TERM};
 
-static void create_sec_context(bool use_tpm, OSSL_LIB_CTX*& out_libctx, SSL_CTX*& out_ctx) {
-    OSSL_LIB_CTX* libctx = OSSL_LIB_CTX_new();
-
-    if (libctx == nullptr) {
-        EVLOG_AND_THROW(std::runtime_error("Unable to create ssl lib ctx."));
-    }
-
-    out_libctx = libctx;
-
-    if (use_tpm) {
-        OSSL_PROVIDER* prov_tpm2 = nullptr;
-        OSSL_PROVIDER* prov_default = nullptr;
-
-        if ((prov_tpm2 = OSSL_PROVIDER_load(libctx, "tpm2")) == nullptr) {
-            EVLOG_AND_THROW(std::runtime_error("Could not load provider tpm2."));
-        }
-
-        if (!OSSL_PROVIDER_self_test(prov_tpm2)) {
-            EVLOG_AND_THROW(std::runtime_error("Could not self-test provider tpm2."));
-        }
-
-        if ((prov_default = OSSL_PROVIDER_load(libctx, "default")) == nullptr) {
-            EVLOG_AND_THROW(std::runtime_error("Could not load provider default."));
-        }
-
-        if (!OSSL_PROVIDER_self_test(prov_default)) {
-            EVLOG_AND_THROW(std::runtime_error("Could not self-test provider default."));
-        }
-    }
-
-    const SSL_METHOD* method = SSLv23_client_method();
-    SSL_CTX* ctx = SSL_CTX_new_ex(libctx, nullptr, method);
-
-    if (ctx == nullptr) {
-        EVLOG_AND_THROW(std::runtime_error("Unable to create ssl ctx."));
-    }
-
-    out_ctx = ctx;
-}
-
-void WebsocketTlsTPM::tls_init() {
-    SSL_CTX* ctx = nullptr;
-
-    if (auto* data = conn_data.get()) {
-        ctx = data->sec_context.get();
-    }
-
-    if (nullptr == ctx) {
-        EVLOG_AND_THROW(std::runtime_error("Invalid SSL context!"));
-    }
-
+void WebsocketTlsTPM::tls_init(SSL_CTX* ctx, const char* path_chain, const char* path_key, bool tpm_key,
```

Comment:
```suggestion
void WebsocketTlsTPM::tls_init(SSL_CTX* ctx, const std::string& path_chain, const std::string& path_key, bool tpm_key,
```

and then do the conversion to const char* further down

Reply:
updated